docs/installation-guide.md

```diff
@@ -1,91 +1,621 @@
+curl -fsSL \
+    "${RAW_ADF_URL}/make/latest/Makefile" \
+    -o ./Makefile.new
+mv ./Makefile.new ./Makefile
+```
+
+Make sure to run `make clean` next.
+
+### 3.4. Build ADF
+
+Build the ADF version with this command next:
+
+```bash
+make
+```
+
+If this fails, run `make clean` first and try again.
+
+### 3.5. Notes on a First-Time Deployment
+
+If you are deploying ADF for the first time, it is recommended to read this
+section carefully and determine which parameters are required for your specific
+use-case.
+
+For example, if you have no AWS Organization or dedicated deployment account
+yet, you can enter an account name and email address and ADF will create you
+an AWS Organization, the deployment organization unit (OU), along with an
+AWS Account that will be used to house deployment pipelines throughout your
+Organization.
+
+If you already have an AWS Account you want to use as your deployment
+account you can specify its Account ID in the parameter
+`DeploymentAccountId` and leave the `DeploymentAccountName` plus
+`DeploymentAccountEmail` empty.
+
+**AWS Control Tower-specific Note:**
+If you use AWS Control Tower, we recommend to create the deployment AWS
+Account via the account vending feature of AWS Control Tower.
+
+It is **MANDATORY**, that your designated deployment AWS Account resides in
+the `deployment` (case-sensitive!) OU. This cannot be changed currently.
+Also make sure there is only one AWS Account in the deployment organization
+unit. Without one, or with more than one the ADF deployment will fail!
+
+The `DeploymentAccountMainRegion` parameter asks for the region that
+will host your deployment pipelines and would-be considered your main AWS
+region.
+
+In the `DeploymentAccountTargetRegions` section of the parameters
+enter a list of AWS Regions that you might want to deploy your resources
+or applications into via AWS CodePipeline *(this can be updated later)*.
+
+When deploying ADF for the first time, part of the installation process will
+automatically create an AWS CodeCommit repository in the management AWS Account
+within the `us-east-1` region. It will also make the initial commit to the
+default branch of this repository with a default set of examples that act as a
+starting point to help define the AWS Account bootstrapping processes for your
+Organization.
+
+Part of the questions that follow will end up in the initial commit into the
+repository. These are passed directly the `adfconfig.yml` file prior to it
+being committed.
+
+### 3.6. Notes on Updating ADF
+
+ADF releases follow the [semantic version number schems](https://semver.org/).
+This implies that breaking changes would require a major version upgrade.
+Hence if you upgrade from `2.x` to `3.x`, it is recommended to check the
+[release notes](https://github.com/awslabs/aws-deployment-framework/releases)
+for details and the breaking changes that it introduced.
+
+### 3.7. Deploy ADF
+
+Next, to deploy ADF, run:
+
+```bash
+make deploy
```

Comment:
I got stuck at this part because it can't locate the template.yml

```sh
Error: Template file not found at /Users/javy/github/aws-deployment-framework/template.yml
make: *** [sam_deploy] Error 1
``` 

Didn't see any prior errors. Where does it move the `template.yml` from `src`?

Reply:
Interesting, did you run `make` before? And did that succeed completely?

in the `sam deploy` step, it should automatically use the `.aws-sam/build/template.yaml`.
Can you check if that exists?

Reply:
Looks like `make version_number` was failing because I don't have `yq` installed and for some reason it's not installing it. After installing that through brew I get a little further, but still fail on the `make version_number` (not sure why).

```
make: *** [version_number] Error 1
```

I also face intermittent issues with this docker command which is part of the make file:

```sh
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```

It randomly hangs, a restart temporarily fixes it. May need some more time with this. Other containers seem to run fine. I do have a modified docker install so it might be a my machine issue.